skrub/_agg_joiner.py

```diff
@@ -145,44 +128,166 @@ class AggJoiner(BaseEstimator, TransformerMixin):
     """
 
     def __init__(
         self,
         aux_table,
         *,
-        aux_key,
-        main_key,
+        key=None,
+        main_key=None,
+        aux_key=None,
         cols=None,
         operation=None,
-        suffix=None,
+        suffix="",
     ):
         self.aux_table = aux_table
+        self.key = key
+        self.main_key = main_key
         self.aux_key = aux_key
         self.cols = cols
-        self.main_key = main_key
         self.operation = operation
         self.suffix = suffix
 
+    def _check_dataframes(self, X, aux_table):
+        """Check dataframes input types.
+
+        Raises an error if frames aren't both Pandas or Polars dataframes,
+        or if there is a Polars lazyframe. Alternatively, allows `aux_table` to be "X".
+
+        Parameters
+        ----------
+        X : DataframeLike
+            The main table to augment.
+        aux_table : DataframeLike or "X"
+            The auxiliary table.
+        """
+        if type(aux_table) == str:
+            if aux_table == "X":
+                return X, deepcopy(X)
+            else:
+                raise ValueError("'aux_table' must be a dataframe or 'X'.")
+        # Polars lazyframes will raise an error here.
+        if not hasattr(X, "__dataframe__"):
+            raise TypeError(f"'X' must be a dataframe, got {type(X)}.")
+        if not hasattr(aux_table, "__dataframe__"):
+            raise TypeError(
+                f"'aux_table' must be a dataframe or 'X', got {type(aux_table)}."
+            )
+
+        if (is_pandas(X) and not is_pandas(aux_table)) or (
+            is_polars(X) and not is_polars(aux_table)
+        ):
+            raise TypeError(
+                "'X' and 'aux_table' must be of the same type, got"
+                f"{type(X)} and {type(aux_table)}"
+            )
+
+        return X, aux_table
+
+    def _check_cols(self):
+        """Check `cols` to aggregate.
+
+        If None, `cols` are all columns from `aux_table`, except `aux_key`.
+
+        Returns
+        -------
+        cols
+            1-dimensional array of columns on which to perform aggregation.
+        """
+        cols = atleast_1d_or_none(self.cols)
+        # If no cols provided, all columns but `aux_key` are used.
+        if len(cols) == 0:
+            cols = list(set(self._aux_table.columns) - set(self._aux_key))
+        if not all([col in self._aux_table.columns for col in cols]):
+            raise ValueError("All 'cols' must be present in 'aux_table'.")
+        return cols
+
+    def _check_operation(self):
+        """Check operation input.
+
+        Returns
+        -------
+        operation
+            1-dimensional array of operations to perform on columns.
+        """
+        operation = atleast_1d_or_none(self.operation)
+        if len(operation) == 0:
+            operation = ["mean", "mode"]
+        return operation
+
+    def _check_suffix(self):
```

Comment:
I'm not sure we need this check, do you forsee this being a frequent user error?

Reply:
For the len check I like it, I think it could be a frequent mistake when using many tables
For the type check I could remove it. It actually works with more than strings since we use `f"{col}{suffix}"` in `px.aggregate`

Reply:
That's true, as you were saying IRL we could use something like `[{"table": df, "aux_key":..., "main_key": ...}, {"table: df2, "aux_key": ..., ...]`, we'll explore this for a future PR !